pom.xml

```diff
@@ -16,13 +16,11 @@
   <url>https://github.com/jenkinsci/promoted-builds-plugin</url>
 
   <properties>
-    <revision>3.12</revision>
+    <revision>4.0</revision>
     <changelist>-SNAPSHOT</changelist>
     <gitHubRepo>jenkinsci/promoted-builds-plugin</gitHubRepo>
-    <!-- Last version before https://jenkins.io/security/advisory/2018-08-15/ which causes instabilities
-         due to Promotion#getTarget(). And we cannot just remove it without breaking binary compatibility
-         (see JENKINS-59704, JENKINS-59773 and so on for the fallout in 3.4)-->
-    <jenkins.version>2.121.1</jenkins.version>
+    <bom>2.289.x</bom>
+    <jenkins.version>2.289.3</jenkins.version>
```

Comment:
why the `.3` here - generally not a good idea to use a not `.1` LTS version (there is a discussion about this somewhere around the bom (caused by config-as-code I think)